tcms/kiwi_auth/views.py

```diff
@@ -27,65 +29,53 @@ class PasswordResetView(views.PasswordResetView):
-            # send a signal that new user has been registered
-            USER_REGISTERED_SIGNAL.send(sender=form.__class__,
-                                        request=request,
-                                        user=new_user)
-
-            # Send confirmation email to new user
-            if settings.DEFAULT_FROM_EMAIL and settings.AUTO_APPROVE_NEW_USERS:
-                form.send_confirm_mail(request, activation_key)
-
-                messages.add_message(
-                    request,
-                    messages.SUCCESS,
-                    _('Your account has been created, please check your mailbox for confirmation')
-                )
-            else:
-                messages.add_message(
-                    request,
-                    messages.WARNING,
-                    _('Your account has been created, but you need an administrator to activate it')
-                )
-                messages.add_message(
-                    request,
-                    messages.INFO,
-                    _('Following is the administrator list')
-                )
-
-                # super-users can approve others
-                for user in User.objects.filter(is_superuser=True):
-                    messages.add_message(
-                        request,
-                        messages.INFO,
-                        '<a href="mailto:{}">{}</a>'.format(user.email,
-                                                            user.get_full_name() or user.username)
-                    )
-
-                # site admins should be able to do so too
-                for name, email in settings.ADMINS:
-                    messages.add_message(
-                        request,
-                        messages.WARNING,
-                        '<a href="mailto:{}">{}</a>'.format(email, name)
-                    )
-
-            return HttpResponseRedirect(reverse('core-views-index'))
-    else:
-        form = forms.RegistrationForm()
-
-    context_data = {
-        'form': form,
-    }
-    return render(request, 'registration/registration_form.html', context_data)
+    template_name = 'registration/registration_form.html'
+    form_class = RegistrationForm
+    success_url = reverse_lazy('core-views-index')
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(data=request.POST, files=request.FILES)
+        if not form.is_valid():
+            return None
+
+        new_user = form.save()
+        activation_key = form.set_activation_key()
+        # send a signal that new user has been registered
+        USER_REGISTERED_SIGNAL.send(sender=form.__class__, request=request, user=new_user)
+
+        # Send confirmation email to new user
+        if settings.DEFAULT_FROM_EMAIL and settings.AUTO_APPROVE_NEW_USERS:
+            form.send_confirm_mail(request, activation_key)
+
+            message = 'Your account has been created, please check your mailbox for confirmation'
```

Comment:
`_()` must be wrapped around the string, not the variable. Otherwise `make messages` will remove these strings from translation files.

Reply:
@atodorov I see. Fixed here.

Reply:
@atodorov , what's the best way to do concatenation of translated strings?  I've tried to find such, but couldn't...

I want to do something like:
```python
        if not form.is_valid():
            words = ['Validation error. Invalid: ']

            if ('username' in form.errors):
                words.append('user name')
                if (len(form.errors) > 1):
                    words.append('and')

            if ('password2' in form.errors):
                words.append('password')

            msg = _(' '.join(words))
            messages.add_message(request, messages.ERROR, msg)
```

It's a headache. Because when didi in a way that is translatable the code-climate complaint about duplication:
```python
        if ('username' in form.errors):
            msg = _('Invalid user name. Please try again')
            messages.add_message(request, messages.ERROR, msg)

        if ('password2' in form.errors):
            msg = _('Invalid password. Please try again')
            messages.add_message(request, messages.ERROR, msg)
```

Now the only way that left atm is:
```python
        msg = _('Invalid username or password. Please try again')
        messages.add_message(request, messages.ERROR, msg)
```
which is not very user friendly...

Reply:
```
 if not form.is_valid():
            words = ['Validation error. Invalid: ']

            if ('username' in form.errors):
                words.append('user name')
                if (len(form.errors) > 1):
                    words.append('and')

            if ('password2' in form.errors):
                words.append('password')

            msg = _(' '.join(words))
            messages.add_message(request, messages.ERROR, msg)
```

you don't need to do anything like that. when the form is invalid it contains all the validation errors in itself, which the template knows how to display if it wants to.